### PR TITLE
Remove the borrow check::nll submodule

### PR DESCRIPTION
**Pull request by mark-i-m**
NLL is the only borrow checker now, so no need to have a separate submodule.

@rustbot modify labels: +S-blocked

Waiting on #66815

### CODE REVIEW

**Comment by mark-i-m** [comment]
cc @eddyb

**Comment by mark-i-m** [comment]
Also, just grepping, it looks like there are a lot of context structs in use in the `rustc_mir::borrow_check` module. Can any of these be merged (I suspect at least the region error reporting stuff can be)?

```
MirBorrowckCtxt               // from rustc_mir::borrow_check
BorrowCheckContext            // from rustc_mir::borrow_check::type_check
LivenessContext               // from rustc_mir::borrow_check::type_check
RegionInferenceContext        // from rustc_mir::borrow_check::region_infer
ErrorReportingCtx             // from rustc_mir::borrow_check::diagnostics:region_errors
RegionErrorNamingCtx          // from rustc_mir::borrow_check::diagnostics::region_naming

// From rustc::mir::visit
TyContext
MutatingUseContext
NonMutatingUseContext
NonUseContext
PlaceContext

// From rustc::infer
InferCtxt

// From rustc::ty
TyCtxt
```

**Comment by matthewjasper** [comment]
`ErrorReportingCtx` can be replaced with `MirBorrowckCtxt` by not reporting those errors until the `MirBorrowckCtxt` is constructed.

File: src/librustc_mir/borrow_check/mod.rs [hunk 1]
```diff
@@ -47,20 +46,30 @@ use self::location::LocationTable;
...
+mod member_constraints;
+
+crate mod constraints;
+crate mod universal_regions;
+crate mod type_check;
+crate mod region_infer;
+crate mod borrow_set;
+crate mod place_ext;
+crate mod places_conflict;
+crate mod nll;
```
Comment (matthewjasper) [inline src/librustc_mir/borrow_check/mod.rs]: Can you check which of these need to be `crate` visible?

**Comment by mark-i-m** [comment]
Rebased, will possibly push more commits soon

**Comment by mark-i-m** [comment]
@matthewjasper Done!

**Comment by matthewjasper** [comment]
r=me with merge conflicts fixed

**Comment by mark-i-m** [comment]
Rebased

**Comment by mark-i-m** [comment]
@matthewjasper Rebased. It looks like there is nothing older in the bors queue that touches the borrowck, so hopefully there won't be any more conflicts...

**Comment by mark-i-m** [comment]
Rebased again. Would it be possible to get p=1 and avoid further rebasing please?

**Pull request merged**